sympy/stats/frv_types.py

```diff
@@ -646,3 +650,204 @@ def Rademacher(name):
 
     """
     return rv(name, RademacherDistribution)
+
+class IdealSolitonDistribution(SingleFiniteDistribution):
+    _argnames = ('k',)
+
+    @staticmethod
+    def check(k):
+         _value_check((k.is_integer, k.is_positive),
+                    "'k' must be a positive integer.")
+
+    @property
+    def low(self):
+        return S.One
+
+    @property
+    def high(self):
+        return self.k
+
+    @property
+    def set(self):
+        return set(list(range(1,self.k+1)))
```

Comment:
This should be changed to sympy's `Range` as python's `range` will not work for symbolic k